Use `mshots` service to generate pattern screenshots

ryelle opened the pull request:
See https://github.com/WordPress/pattern-directory/pull/521#discussion_r979500622 — This swaps out the iframes in the pattern grid in favor of an mshots request. This does improve the number and overall size of requests, but it doesn't work on local sites.

**Stats** for loading the homepage (and scrolling down to load all pattern previews)

`trunk`: 416 requests, 30.75 MB / 9.35 MB transferred.
`try/mshots`: 177 requests, 5.91 MB / 3.28 MB transferred.

I forked [the ScreenShotImg component from `wporg-mu-plugins`](https://github.com/WordPress/wporg-mu-plugins/blob/trunk/mu-plugins/blocks/screenshot-preview/src/screenshot.js) so it could be used here, and added a few more props.

Fixes #349

### Screenshots

| Before | After |
|--------|------|
| <video src="https://user-images.githubusercontent.com/541093/194179167-04a45eed-9ecd-4daa-bacb-b8d4058762eb.mp4"> | <video src="https://user-images.githubusercontent.com/541093/194179174-d3c48a65-8580-4d06-bcd2-616d8b64d68f.mp4"> |

On local sites it just returns this 403 image. This is not great, but I'm not sure if this should be a blocker.

![Screen Shot 2022-10-05 at 19 08 47](https://user-images.githubusercontent.com/541093/194180097-cdfcd059-66f3-4296-b1e9-12ba441efc23.png)

### How to test the changes in this Pull Request:

1. Apply this to a sandbox
2. View assorted pattern grids — categories, my patterns, etc
3. The small pattern previews should look correct & match the larger preview on the pattern detail page

ryelle committed:
Use `mshots` service to generate pattern screenshots

StevenDufresne commented on public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/canvas.js:
```diff
@@ -40,32 +39,27 @@ export default function ( { url } ) {
-				src={ shouldLoad ? url : '' }
+				size={ { width: VIEWPORT_WIDTH, height: getCardFrameHeight( VIEWPORT_WIDTH ) } }
+				isReady={ shouldLoad }
+				src={ url }
```
Can patterns be updated? 

If so you'll want to introduce a way to break the cache so a new mShot is generated.

ryelle replied:
Yeah, they can, good catch — I added the pattern's modified date as part of the URL.

StevenDufresne commented on public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/screenshot.js:
```diff
@@ -0,0 +1,117 @@
+	if ( isLoading ) {
+		return (
+			<div className={ `${ className } is-loading` } style={ style }>
+				<Spinner style={ { width: '33%', height: '33%' } } />
```
The spinner feels a little bit large, but that's personal preference.

ryelle replied:
Fair enough - the default size was very tiny, and this wasn't really chosen with any intention. Swapped it out for 2x default size.

<img width="336" alt="Screen Shot 2022-10-06 at 12 27 49 PM" src="https://user-images.githubusercontent.com/541093/194367815-7bfc6752-eb5b-46ea-9adf-066c50adf623.png">

ryelle committed:
Remove the `Disabled` wrapper

Since this isn't an iframe anymore, there isn't anything possibly-interactive we need to disable.

ryelle committed:
Update the spinner size

ryelle committed:
Move the size values to Screenshot

Set both image width and viewport size, so the screenshot is taken at 1200px wide, but then resized down to a 600px image before being sent back to the browser.

ryelle committed:
Add the modified date to the URL

This will break the image cache if the pattern is updated

StevenDufresne commented:
LGTM 👍.

ryelle commented on public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/index.js:
```diff
@@ -35,9 +34,7 @@ function PatternThumbnail( { pattern, showAvatar, showOptions } ) {
-					<Disabled>
-						<Canvas url={ addQueryArgs( pattern.link, { view: true } ) } />
-					</Disabled>
+					<Canvas url={ addQueryArgs( pattern.link, { view: true, modified: pattern.modified_gmt } ) } />
```
`modified` will break the cache when a pattern is modified, but there should probably also be a site-wide cachebuster in case we update styles in the pattern preview endpoint.

StevenDufresne replied:
I wouldn't be opposed to a manual cache break if it's really needed in short order.

ryelle committed:
Add global "thumnailVersion" cachebuster for mshots image cache

ryelle committed:
Fetch the screenshot from the production site if this is a local env

This fixes most of the "403" errors when using the exported data, since the same patterns exist in production. The exception is any locally-created patterns, which now show a 404 page. Still, this is an improvement over errors everywhere locally.
